api/handlers.go

```diff
@@ -2,32 +2,153 @@ package api
+	"strconv"
 
 	"github.com/NebulousLabs/skynet-accounts/database"
+	"github.com/NebulousLabs/skynet-accounts/metafetcher"
+	"gitlab.com/NebulousLabs/errors"
 
 	"github.com/julienschmidt/httprouter"
-	"gitlab.com/NebulousLabs/errors"
 )
 
-// userHandlerGET returns information about an existing user and create it if it
+// userHandler returns information about an existing user and create it if it
 // doesn't exist.
-func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+func (api *API) userHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	WriteJSON(w, u)
+}
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	offset, _ := strconv.Atoi(ps.ByName("offset"))
+	limit, _ := strconv.Atoi(ps.ByName("limit"))
+	ups, err := api.staticDB.UploadsByUser(req.Context(), *u, offset, limit)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+	}
+	WriteJSON(w, ups)
+}
+
+// userDownloadsHandler returns all downloads made by the current user.
+func (api *API) userDownloadsHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	offset, _ := strconv.Atoi(ps.ByName("offset"))
+	limit, _ := strconv.Atoi(ps.ByName("limit"))
+	ups, err := api.staticDB.DownloadsByUser(req.Context(), *u, offset, limit)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+	}
+	WriteJSON(w, ups)
+}
+
+// trackUploadHandler registers a new upload in the system.
+func (api *API) trackUploadHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 	sub, _, _, err := tokenFromContext(req)
 	if err != nil {
 		WriteError(w, err, http.StatusInternalServerError)
 		return
 	}
-	u, err := api.staticDB.UserBySub(req.Context(), sub)
-	if err != nil && !errors.Contains(err, database.ErrUserNotFound) {
+	sl := ps.ByName("skylink")
+	if sl == "" {
+		WriteError(w, errors.New("missing parameter 'skylink'"), http.StatusBadRequest)
+		return
+	}
+	skylink, err := api.staticDB.Skylink(req.Context(), sl)
+	if err == database.ErrInvalidSkylink {
```

Comment:
Might be good to start using `errors.Contains` from the beginning to avoid having to refactor code in the future.